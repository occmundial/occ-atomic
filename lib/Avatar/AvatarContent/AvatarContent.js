import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../../Icon';
import colors from '../../tokens/colors.json';

const getInitials = (name) => {
    if (name && name.trim()) {
        const [firstName, lastName] = name
            .trim()
            .replace(/\s+/g, ' ')
            .toUpperCase()
            .split(' ');
        return `${firstName?.charAt(0)}${lastName?.charAt(0)}`;
    }
    return null;
  };

const AvatarContent = ({
    classes,
    photo,
    name,
    size,
    onEdit,
    onClick,
    disabled
}) => {
    const initials = getInitials(name);
    return (
        <div className={classes.wrap}>
            <div
                className={`${classes.cont} ${
                    onClick ? ` ${classes.clickable}` : ''}${
                    onEdit ? ` ${classes.editable}` : ''}${
                    disabled ? ` ${classes.disabled}` : ''}${
                    photo ? ` ${classes.photo}` : ''}${
                    photo && (onClick || onEdit) ? ` ${classes.hoverPhoto}` : ''}
                `}
                style={
                    photo
                        ? {
                            background: `url('${photo}') no-repeat center center / cover`
                        }
                        : null
                }
            >
            {!photo && !initials && (
                <Icon
                    iconName="person"
                    className={classes.person}
                    colors={[
                    disabled
                        ? colors.icon.default.disabled
                        : colors.icon.brand.default
                    ]}
                    size={24}
                    style={size ? { transform: `scale(${size / 40})` } : {}}
                />
            )}
            {initials && (
                <span
                    className={`${classes.initials} ${
                        disabled ? classes.disabledInitials : ''
                    }`}
                    style={size ? { transform: `scale(${size / 40})` } : {}}
                >
                    {initials}
                </span>
            )}
            <div className={classes.overlay}>
                {onEdit && !disabled && (
                    <Icon
                        iconName="camera"
                        className={classes.camera}
                        colors={[
                            disabled
                            ? colors.icon.default.disabled
                            : colors.icon.inverse.default
                        ]}
                        size={24}
                        style={size ? { transform: `scale(${size / 40})` } : {}}
                    />
                )}
            </div>
        </div>
    </div>
)};

AvatarContent.propTypes = {
    classes: PropTypes.object,
    photo: PropTypes.string,
    gender: PropTypes.string,
    name: PropTypes.string,
    size: PropTypes.number,
    onEdit: PropTypes.func,
    onClick: PropTypes.func,
    disabled: PropTypes.bool
};

export default AvatarContent;