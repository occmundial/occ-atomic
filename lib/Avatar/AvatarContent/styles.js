import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
