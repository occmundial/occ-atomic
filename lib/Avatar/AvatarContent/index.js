import AvatarContent from './AvatarContent';

export default AvatarContent;
