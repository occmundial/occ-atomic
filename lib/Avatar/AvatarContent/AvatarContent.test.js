import React from 'react';
import { mount } from 'enzyme';
import { JssProvider, SheetsRegistry } from 'react-jss';

import AvatarContent from './AvatarContent';

let sheets;

describe('AvatarContent', () => {

    beforeEach(() => sheets = new SheetsRegistry());
    afterEach(() => sheets = null);

    it('matches the snapshot', () => {
        const wrapper = mount(
            <AvatarContent name="John Doe" />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('returns the first two capital letters of the name', () => {
        const wrapper = mount(
            <JssProvider registry={sheets}>
                <AvatarContent name="John Doe" />
            </JssProvider>
        );
        const { classes } = sheets.registry[0];
        expect(wrapper.find(`.${classes.capital} span`).text()).toEqual('JD');
    });

    it('shows the right gender icon', () => {
        let wrapper = mount(
            <JssProvider registry={sheets}>
                <AvatarContent gender="m" />
            </JssProvider>
        );
        let classes = sheets.registry[0].classes;
        expect(wrapper.find(`.${classes.m}`).length).toBe(1);
        expect(wrapper.find(`.${classes.f}`).length).toBe(0);
        wrapper = mount(
            <JssProvider registry={sheets}>
                <AvatarContent gender="f" />
            </JssProvider>
        );
        classes = sheets.registry[0].classes;
        expect(wrapper.find(`.${classes.m}`).length).toBe(0);
        expect(wrapper.find(`.${classes.f}`).length).toBe(1);
    });

    it('renders the photo as a background', () => {
        const wrapper = mount(
            <JssProvider registry={sheets}>
                <AvatarContent photo="http://placehold.it/70x70" />
            </JssProvider>
        );
        const { classes } = sheets.registry[0];
        expect(wrapper.find(`.${classes.cont}`).prop('style')).toHaveProperty('background', "url('http://placehold.it/70x70') no-repeat center center / cover");
    });

    it('changes the font size', () => {
        const wrapper = mount(
            <JssProvider registry={sheets}>
                <AvatarContent name="John Doe" size={50} />
            </JssProvider>
        );
        const { classes } = sheets.registry[0];
        const styles = wrapper.find(`.${classes.capital}`).prop('style');
        expect(styles).toHaveProperty('fontSize', '20px');
        expect(styles).toHaveProperty('lineHeight', '50px');
    });

});
