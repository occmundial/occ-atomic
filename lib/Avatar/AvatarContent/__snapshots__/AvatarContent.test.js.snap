// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarContent matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`AvatarContent styles matches the snapshot 1`] = `
Object {
  "camera": Object {
    "gridColumnStart": 1,
    "gridRowStart": 1,
    "opacity": 0,
    "transition": "all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s",
  },
  "clickable": Object {
    "&:active": Object {
      "background": "#ABBDE4",
    },
    "&:hover": Object {
      "background": "#CAD5EE",
    },
    "transition": "all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s",
  },
  "cont": Object {
    "alignItems": "center",
    "background": "#E9EEF8",
    "borderRadius": "9999px",
    "display": "grid",
    "height": "100%",
    "justifyContent": "center",
    "left": "0",
    "overflow": "hidden",
    "position": "absolute",
    "top": "0",
    "width": "100%",
  },
  "disabled": Object {
    "& $camera": Object {
      "opacity": 1,
    },
    "background": "#EDEDF1",
    "pointerEvents": "none",
  },
  "disabledInitials": Object {
    "color": "#878A9F",
  },
  "editable": Object {
    "&:active $overlay": Object {
      "background": "rgba(8, 13, 57, 0.4)",
    },
    "&:hover $camera": Object {
      "opacity": 1,
    },
    "&:hover $overlay": Object {
      "background": "rgba(8, 13, 57, 0.3)",
    },
  },
  "hoverPhoto": Object {
    "&:active $overlay": Object {
      "background": "rgba(8, 13, 57, 0.4)",
    },
    "&:hover $overlay": Object {
      "background": "rgba(8, 13, 57, 0.3)",
    },
  },
  "initials": Object {
    "color": "#083CAE",
    "display": "inline-block",
    "font": "400 16px/1.5 'OccText', sans-serif",
    "gridColumnStart": "1",
    "gridRowStart": "1",
    "textAlign": "center",
  },
  "overlay": Object {
    "alignItems": "center",
    "border": "1px solid transparent",
    "borderRadius": "50%",
    "display": "grid",
    "height": "100%",
    "justifyContent": "center",
    "left": "0",
    "position": "absolute",
    "top": "0",
    "transition": "all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s",
    "width": "100%",
  },
  "person": Object {
    "gridColumnStart": 1,
    "gridRowStart": 1,
    "transition": "all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s",
  },
  "photo": Object {
    "border": "1px solid #EDEDF1",
  },
  "wrap": Object {
    "height": "100%",
    "position": "relative",
    "width": "100%",
  },
}
`;

exports[`AvatarContentJSS matches the snapshot 1`] = `ShallowWrapper {}`;
